addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/ContentSecurityPolicyMissingScanRule.java

```diff
@@ -111,12 +109,25 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
                     .setWascId(15) // WASC-15: Application Misconfiguration
                     .raise();
         }
+        // ... or as a warning at LOW thresshold if one of the obsolete headers is found
+        if (AlertThreshold.LOW.equals(this.getAlertThreshold())
+                && (xCspHeaderFound || xWebKitHeaderFound)) {
+            newAlert()
+                    .setRisk(Alert.RISK_INFO)
+                    .setConfidence(Alert.CONFIDENCE_HIGH)
+                    .setDescription(getAlertAttribute("desc"))
+                    .setSolution(getAlertAttribute("soln"))
+                    .setReference(getAlertAttribute("refs"))
+                    .setCweId(693) // CWE-693: Protection Mechanism Failure
+                    .setWascId(15) // WASC-15: Application Misconfiguration
+                    .raise();
+        }
 
         if (cspROHeaderFound) {
             newAlert()
                     .setName(getAlertAttribute("ro.name"))
                     .setRisk(Alert.RISK_INFO)
-                    .setConfidence(Alert.CONFIDENCE_HIGH)
+                    .setConfidence(Alert.CONFIDENCE_MEDIUM)
```

Comment:
Why is there any debate whether the header was found or not?

Reply:
There's no debate; that was not even the line I intended to change, so I'll back this out.